npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -193,27 +191,62 @@ def handle_npm_updater_error(error, lockfile)
              error.message.include?("Non-registry package missing package") ||
              error.message.include?("Cannot read property 'match' of ") ||
              error.message.include?("Invalid tag name")
-            # This happens if a new version has been published that relies on
-            # subdependencies that have not yet been published.
-            raise if resolvable_before_update?(lockfile)
+            unless resolvable_before_update?(lockfile)
+              raise_resolvability_error(error, lockfile)
+            end
 
-            raise_resolvability_error(error, lockfile)
+            # TODO: Move this logic to the version resolver and check if a new
+            # version and all of its subdependencies are resolvable
+
+            # Make sure the error in question matches the current list of
+            # dependencies or matches an existing scoped package, this handles
+            # the case where a new version (e.g. @angular-devkit/build-angular)
+            # relies on a added dependency which hasn't been published yet under
+            # the same scope (e.g. @angular-devkit/build-optimizer)
+            #
+            # This seems to happen when big monorepo projects publish all of
+            # their packages sequentially, which might take enough time for
+            # Dependabot to hear about a new version before all of its
+            # dependencies have been published.
+            names = dependencies.map { |dep| dep.name.split("/").first }
+            if names.any? { |name| error.message.include?(name) }
```

Comment:
`error.message.match?(/Regexp.quote(name)[/@]/)` so as the name has to either match a dependency's scope or the end of its name.